src/Data/Scope.php

```diff
@@ -0,0 +1,67 @@
+<?php
+namespace LORIS\Data;
+
+/**
+ * A Scope is an enumeration class which represents the scope
+ * that a piece of data may apply to in LORIS.
+ *
+ * The Scope class is final because the list of enumeration types
+ * can not be dynamically extended without modifying all places
+ * that must deal with the enumeration options.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+final class Scope implements \JsonSerializable
+{
+    // Valid scopes for data to apply to.
+    const CANDIDATE = 1;
+    const SESSION   = 2;
+
+    /**
+     * The value of the current scope instance
+     */
+    protected $scope;
+
+    /**
+     * Constructs a Scope object. $scope should be a class constant
+     * to construct the scope for, not an int literal.
+     *
+     * @param int $scope The scope
+     */
+    public function __construct(int $scope)
+    {
+        switch ($scope) {
+            case self::CANDIDATE: // fallthrough
```

Comment:
Shouldn't there be a `break;` for case self::CANDIDATE:

Reply:
no, it intentionally falls through (hence the line comment)

Reply:
Oh I thought break is necessary to avoid passing through to the code under the next label

Reply:
it does intentionally pass through to the code under the next label (which just sets the internal value to the variable $scope)